packages/@sfx/core/test/unit/common/utils/core.test.ts

```diff
@@ -0,0 +1,315 @@
+          },
+        },
+        {
+          metadata: {
+            name: 'n',
+            depends: ['b'],
+          },
+        },
+      ]
+
+      const missing = calculateMissingDependencies(plugins, registry);
+
+      expect(missing).to.deep.equal([]);
+    });
+
+    it('should return no missing dependencies when there are no dependencies', () => {
+      const registry = {
+        a: 'aa',
+        b: 'bb',
+        c: 'cc',
+      };
+      const plugins: any = [
+        {
+          metadata: {
+            name: 'm',
+            depends: [],
+          },
+        },
+        {
+          metadata: {
+            name: 'n',
+            depends: [],
+          },
+        },
+      ];
+
+      const missing = calculateMissingDependencies(plugins, registry);
+
+      expect(missing).to.deep.equal([]);
+    });
+
+    it('should return no missing dependencies when dependencies are satisfied by the new plugins', () => {
+      const registry = {
+        c: 'cc',
+      };
+      const plugins: any = [
+        {
+          metadata: {
+            name: 'a',
+            depends: ['b', 'c'],
+          },
+        },
+        {
+          metadata: {
+            name: 'b',
+            depends: ['c'],
+          },
+        },
+      ];
+
+      const missing = calculateMissingDependencies(plugins, registry);
+
+      expect(missing).to.deep.equal([]);
+    });
+
+    it('should return no missing dependencies for satisfied circular dependencies', () => {
+      const registry = {};
+      const plugins: any = [
+        {
+          metadata: {
+            name: 'a',
+            depends: ['b'],
+          },
+        },
+        {
+          metadata: {
+            name: 'b',
+            depends: ['c'],
+          },
+        },
+        {
+          metadata: {
+            name: 'c',
+            depends: ['a'],
+          },
+        },
+      ];
+
+      const missing = calculateMissingDependencies(plugins, registry);
+
+      expect(missing).to.deep.equal([]);
+    });
+  });
+
+  describe('registerPlugins()', () => {
+    it('should call the register function of each plugin with the plugin registry', () => {
+      let localRegistryA;
+      let localRegistryB;
+      const plugins: any = [
+        {
+          metadata: { name: 'a' },
+          register: (r) => localRegistryA = r,
+        },
+        {
+          metadata: { name: 'b' },
+          register: (r) => localRegistryB = r,
+        },
+      ];
+      const registry: any = {};
+
+      registerPlugins(plugins, registry);
+
+      expect(localRegistryA).to.be.an('object');
+      expect(localRegistryB).to.be.an('object');
+
+      const cvalue = registry.c = 'cc';
+
+      expect(localRegistryA.c).to.equal(cvalue);
+      expect(localRegistryB.c).to.equal(cvalue);
+    });
+
+    it('should return a map of new plugin keys to exposed values', () => {
+      const valueA = { a: 'a' };
+      const valueB = { b: 'b' };
+      const valueC = { c: 'c' };
```

Comment:
We probably should have considered using varying datatypes for the values above. It makes it look like plugins have to expose objects, which is not true, and I find that distracts from the test.